aten/src/THC/THCTensorIndex.cu

```diff
@@ -17,6 +17,7 @@
 #include <thrust/sort.h>
 #include <algorithm> // for std::min
 #include <c10/macros/Macros.h>
+#include <ATen/WrapDimUtils.h>
```

Comment:
It's just a way of including the file once instead of # of dtypes times.

The way the TH macros work is:
aten/src/TH(C)/TH(C)MyFile.ext includes
aten/src/TH(C)/generic/TH(C)MyFile.ext 1 time for each dtype, setting what e.g. `scalar_t` means.

So in this case it's the generic version that actually uses the functions here, I'm just including it in the non-generic version to simplify the include logic.